rp235x-hal/src/rom_data.rs

```diff
@@ -109,6 +109,526 @@ pub fn rom_data_lookup(tag: RomFnTableCode, mask: u32) -> usize {
+    InvalidArg = -5,
+    /// An address argument was out-of-bounds or was determined to be an address
+    /// that the caller may not access
+    InvalidAddress = -10,
+    /// An address passed to the function was not correctly aligned
+    BadAlignment = -11,
+    /// Something happened or failed to happen in the past, and consequently the
+    /// request cannot currently be serviced.
+    InvalidState = -12,
+    /// A user-allocated buffer was too small to hold the result or working state
+    /// of the function
+    BufferTooSmall = -13,
+    /// The call failed because another bootrom function must be called first.
+    PreconditionNotMet = -14,
+    /// Cached data was determined to be inconsistent with the full version of
+    /// the data it was copied from
+    ModifiedData = -15,
+    /// The contents of a data structure are invalid
+    InvalidData = -16,
+    /// An attempt was made to access something that does not exist; or, a search failed
+    NotFound = -17,
+    /// Modification is impossible based on current state; e.g. attempted to clear
+    /// an OTP bit.
+    UnsupportedModification = -18,
+    /// A required lock is not owned. See Section 5.4.4.
+    LockRequired = -19,
+    /// An unknown error
+    Unknown = -1,
+}
+
+impl From<i32> for BootRomApiErrorCode {
+    fn from(value: i32) -> Self {
+        match value {
+            -4 => Self::NotPermitted,
+            -5 => Self::InvalidArg,
+            -10 => Self::InvalidAddress,
+            -11 => Self::BadAlignment,
+            -12 => Self::InvalidState,
+            -13 => Self::BufferTooSmall,
+            -14 => Self::PreconditionNotMet,
+            -15 => Self::ModifiedData,
+            -16 => Self::InvalidData,
+            -17 => Self::NotFound,
+            -18 => Self::UnsupportedModification,
+            -19 => Self::LockRequired,
+            _ => Self::Unknown,
+        }
+    }
+}
+
+/// This module defines a safe api to access the `get_sys_info` bootrom function
+#[allow(unused)]
+pub mod sys_info_api {
+    use super::BootRomApiErrorCode;
+
+    /// Flags that the `get_sys_info`/ rom function can take
+    #[repr(u32)]
+    pub enum GetSysInfoFlag {
+        /// The flag used to get a chip's unique identifier
+        ChipInfo = 0x0001,
+        /// The flag used to get the critical register's value
+        Critical = 0x0002,
+        /// The flag used to get the current running CPU Architecture
+        CpuInfo = 0x0004,
+        /// The flag used to get flash device info
+        FlashDevInfo = 0x0008,
+        /// The flag used to get the random 128 bit integer generated on boot
+        BootRandom = 0x0010,
+        // Ignore nonce for now since it can't/shouldn't be called anyway?
+        // Nonce = 0x0020,
+        /// The flag used to get boot diagnostic info
+        BootInfo = 0x0040,
+    }
+
+    impl GetSysInfoFlag {
+        const fn buffer_length(&self) -> usize {
```

Comment:
Maybe add a note that the sys info flags are a bitmask and can be OR'd together, but we only ever send one flag at a time so we don't need to add the sizes together.